packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -889,11 +878,47 @@ function simplifyIndexed<T>(
 		context.ctx.logger.warn(`Tried to index un-indexable type ${child.kind}`)
 		return { kind: 'union', members: [] }
 	}
+
+	const symbolMap: { [key: string]: { data?: unknown } } = {}
+	for (const field of child.fields) {
+		if (field.key.kind === 'literal' && field.key.value.kind === 'string') {
+			symbolMap[field.key.value.value] = {
+				data: {
+					typeDef: field.type,
+				} satisfies TypeDefSymbolData,
```

Comment:
It feels weird re-using the `TypeDefSymbolData` for non-dispatcher types tbh, but maybe it's the cleanest solution.

Reply:
An alternative would be to map the type when using a dispatcher, but since dispatcher is the default case, I wanted to try to not do any mapping on the dispatcher.

It probably wouldn't change much though.